### PR TITLE
Port hex command

### LINKED ISSUE #10: Port Hex command

### PR DESCRIPTION
**Pull request by Steffan153**
Closes #10

### CODE REVIEW

**Comment by Steffan153** [comment]
oops that .vscode folder wasn't supposed to get in there

**Comment by jodylecompte** [comment]
Please remove vscode and add to gitignore

File: commands/hex/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const { permissionType, commandType } = require('../../lib/permissions');
+
+module.exports = {
+  name: 'hex',
+  aliases: ['hex'],
+  type: commandType.base.name,
+  permissions: permissionType.user,
+  template: 'hex',
+  async handler({ Discord, client, message }) {
+    let s = message.content.slice(5) || 'random';
```
Comment (drillskibo) [inline commands/hex/index.js:10]: Please change this to the `args` variable, present in [handler functions arguments](https://github.com/PopperDevs/PopperBot/blob/development/index.js#L32). It automatically stripes out the command and gets an array of arguments, splitted by space.
You can eventually join by space again if you are in need of a full "sentence" after the command, or just use `args[0]`.

File: commands/color/hex.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    if (s === 'random') s = Math.random().toString(16).slice(2, 8);
+    if (s.length === 3) s = [...s].map((x) => x + x).join('');
+    return message.channel.send(
+      new Discord.MessageEmbed()
+        .setColor('#' + s)
+        .setTitle('#' + s)
+        .setThumbnail(`https://via.placeholder.com/150/${s}?text=+`)
+        .setTimestamp(message.createdAt)
+        .setFooter(message.author.tag)
```
Comment (drillskibo) [inline commands/color/hex.js:26]: Please change footer to `hex | ${message.author.tag}` for [consistency](https://github.com/PopperDevs/PopperBot/blob/master/commands/info/index.js#L18).

File: commands/color/rgb.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      .map((x) => (+x).toString(16).padStart(2, 0))
+      .join('')
+      .toUpperCase();
+    return message.channel.send(
+      new Discord.MessageEmbed()
+        .setColor('#' + hexCode)
+        .setTitle('RGB ' + args.join(', '))
+        .setThumbnail(`https://via.placeholder.com/150/${hexCode}?text=+`)
+        .setTimestamp(message.createdAt)
+        .setFooter(message.author.tag)
```
Comment (drillskibo) [inline commands/color/rgb.js:37]: Please change footer to `hex | ${message.author.tag}` for [consistency](https://github.com/PopperDevs/PopperBot/blob/master/commands/info/index.js#L18).

File: commands/color/rgb.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const { permissionType, commandType } = require('../../lib/permissions');
+
+module.exports = {
+  name: 'rgb',
+  aliases: ['rgb'],
+  type: commandType.base.name,
+  permissions: permissionType.user,
+  template: 'rgb',
+  async handler({ Discord, args, message }) {
```
Comment (drillskibo) [inline commands/color/rgb.js:9]: Please remove `async` keyword since you are not awaiting anything.

File: commands/color/hex.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+const { permissionType, commandType } = require('../../lib/permissions');
+
+module.exports = {
+  name: 'hex',
+  aliases: ['hex'],
+  type: commandType.base.name,
+  permissions: permissionType.user,
+  template: 'hex',
+  async handler({ Discord, args, message }) {
```
Comment (drillskibo) [inline commands/color/hex.js:9]: Please remove `async` keyword since you are not awaiting anything.

File: commands/color/hex.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+const { permissionType, commandType } = require('../../lib/permissions');
+
+module.exports = {
```
Comment (drillskibo) [inline commands/color/hex.js:3]: Please add parent property to the subCommand object like in [here](https://github.com/PopperDevs/PopperBot/blob/master/commands/commands/new.js#L3) for example.

File: commands/color/rgb.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+const { permissionType, commandType } = require('../../lib/permissions');
+
+module.exports = {
```
Comment (drillskibo) [inline commands/color/rgb.js:3]: Please add parent property to the subCommand object like in [here](https://github.com/PopperDevs/PopperBot/blob/master/commands/commands/new.js#L3) for example.

File: commands/color/rgb.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  name: 'rgb',
+  aliases: ['rgb'],
```
Comment (drillskibo) [inline commands/color/rgb.js:5]: You don't need to put the command name again in aliases. You can leave an empty array if you don't have any.

File: commands/color/index.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+const { permissionType, commandType } = require('../../lib/permissions');
+
+module.exports = {
+  name: 'color',
+  aliases: ['color'],
```
Comment (drillskibo) [inline commands/color/index.js:5]: You don't need to put the command name again in aliases. You can leave an empty array if you don't have any.

File: commands/color/hex.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  name: 'hex',
+  aliases: ['hex'],
```
Comment (drillskibo) [inline commands/color/hex.js:5]: You don't need to put the command name again in aliases. You can leave an empty array if you don't have any.

File: commands/color/hex.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  type: commandType.base.name,
+  permissions: permissionType.user,
+  template: 'hex',
+  async handler({ Discord, args, message }) {
+    let s = args[0];
+    if (s !== 'random' && !/^([0-9a-f]{3}){1,2}$/i.test(s)) {
+      return message.channel.send(
+        new Discord.MessageEmbed()
+          .setColor('#FF9AA2')
+          .setTitle('Invalid usage. Please put a valid hex code in.')
```
Comment (drillskibo) [inline commands/color/hex.js:15]: Please add templating to the error message.

File: commands/color/rgb.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    if (
+      s !== 'random' &&
+      (!/^\d{1,3} \d{1,3} \d{1,3}$/i.test(s) ||
+        !args.every((x) => +x >= 0 && +x <= 255))
+    ) {
+      return message.channel.send(
+        new Discord.MessageEmbed()
+          .setColor('#FF9AA2')
+          .setTitle(
+            'Invalid usage. Please enter a valid RGB code, separated by spaces.'
```
Comment (drillskibo) [inline commands/color/rgb.js:21]: Please add templating to the error message.

File: commands/color/index.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+  handler({ Discord, message, args }) {
+    if (args.length === 1) args.unshift('hex');
+    const subCommand = this.subCommands.get(args[0]);
+
+    if (subCommand) {
+      args.shift();
+      subCommand.handler({ Discord, message, args });
+    } else {
+      message.channel.send(
+        `Incorrect syntax! Correct usage of this command: \`${process.env.PREFIX}${this.template} <hex/rgb> <color>\``
```
Comment (drillskibo) [inline commands/color/index.js:18]: Please use the same embed styling as the subCommands for consistency.

**Pull request merged**